src/test/java/org/gridsuite/actions/server/ContingencyListControllerTest.java

```diff
@@ -0,0 +1,107 @@
+import static org.mockito.BDDMockito.given;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.MediaType.TEXT_PLAIN;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+@RunWith(SpringRunner.class)
+@WebMvcTest(ContingencyListController.class)
+@ContextConfiguration(classes = {ActionsApplication.class})
+public class ContingencyListControllerTest extends AbstractEmbeddedCassandraSetup {
+
+    private static final UUID NETWORK_UUID = UUID.fromString("7928181c-7977-4592-ba19-88027e4254e4");
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ScriptContingencyListRepository contingencyListRepository;
+
+    @Autowired
+    private ContingencyListService contingencyListService;
+
+    @MockBean
+    private NetworkStoreService networkStoreService;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        Network network = EurostagTutorialExample1Factory.create(new NetworkFactoryImpl());
+        given(networkStoreService.getNetwork(NETWORK_UUID, PreloadingStrategy.COLLECTION)).willReturn(network);
+    }
+
+    @Test
+    public void test() throws Exception {
+        String script =
+            "contingency('NHV1_NHV2_1') {" +
+            "     equipments 'NHV1_NHV2_1'" +
+            "}";
+
+        mvc.perform(put("/" + VERSION + "/script-contingency-lists/foo")
+                .content(script)
+                .contentType(TEXT_PLAIN))
+                .andExpect(status().isOk());
+
+        mvc.perform(get("/" + VERSION + "/contingency-lists")
+                .contentType(APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(content().json("[{\"name\":\"foo\",\"script\":\"contingency('NHV1_NHV2_1') {     equipments 'NHV1_NHV2_1'}\"}]"));
+
+        mvc.perform(get("/" + VERSION + "/contingency-lists/bar")
+                .contentType(APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().string(""));
+
+        mvc.perform(get("/" + VERSION + "/contingency-lists/foo")
+                .contentType(APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(content().json("{\"name\":\"foo\",\"script\":\"contingency('NHV1_NHV2_1') {     equipments 'NHV1_NHV2_1'}\"}"));
+
+        mvc.perform(get("/" + VERSION + "/contingency-lists/foo/export")
```

Comment:
Add a test for a contingency script that doesn't need a network ?

Reply:
Should we add a param to the contingency script, telling if a network is needed or not ?

Reply:
The only case we don't need a network is an empty list (so no groovy code). I added a test for this case.